foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,213 @@
+            bookmark_data (BookmarkData, optional): Data for the bookmark when using detailed menu.
+                Required when detailed is True.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            if not is_detailed:
+                self.selenium.find_element(*BookmarkLocators.STAR_BUTTON).click()
+                self.selenium.find_element(*BookmarkLocators.FOLDER_MENU).click()
+                self.selenium.find_element(*BookmarkLocators.OTHER_BOOKMARKS_STAR).click()
+                self.selenium.find_element(*BookmarkLocators.SAVE_BUTTON).click()
+            else:
+                with self.selenium.context(self.selenium.CONTEXT_CHROME):
+                    self.actions.context_click(
+                        self.selenium.find_element(*BookmarkLocators.NAVIGATOR_TOOLBOX)
+                    ).perform()
+                    self.selenium.find_element(*BookmarkLocators.MENU_BAR).click()
+                    self.selenium.find_element(
+                        *BookmarkLocators.MAIN_MENU_BOOKMARK
+                    ).click()
+                    self.actions.context_click(
+                        self.selenium.find_element(*BookmarkLocators.MANAGE_BOOKMARKS)
+                    ).perform()
+                    self.selenium.find_element(*BookmarkLocators.ADD_BOOKMARK).click()
+
+                    frame_element = self.selenium.find_element(
+                        *BookmarkLocators.ADD_BOOKMARK_FRAME
+                    )
+                    self.selenium.switch_to.frame(frame_element)
+                    if bookmark_data is not None:
+                        if bookmark_data["name"]:
+                            self.actions.send_keys(bookmark_data["name"]).perform()
+                        self.actions.send_keys(Keys.TAB).perform()
+
+                        if bookmark_data["url"]:
+                            self.actions.send_keys(
+                                bookmark_data["url"] + Keys.TAB
+                            ).perform()
+
+                        if (tags := bookmark_data["tags"]) is not None:
+                            for tag in tags:
+                                self.actions.send_keys(tag).perform()
+                                self.actions.send_keys(",").perform()
+                            self.actions.send_keys(Keys.TAB).perform()
+
+                        if bookmark_data.get("keyword"):
+                            keyword = bookmark_data["keyword"] or ""
+                            self.actions.send_keys(keyword + Keys.TAB).perform()
+
+                    self.actions.send_keys(
+                        Keys.TAB, Keys.TAB, Keys.TAB, Keys.ENTER
+                    ).perform()
+                    if (
+                        folder := self.selenium.find_element(
+                            *BookmarkLocators.BOOKMARK_FOLDER
+                        )
+                    ) is not None:
```

Comment:
I don't think we need to do this `None` checks. You can remove the `is not None` part and do the same check as well. Also, good use of the walrus operator!